spec/helper/revenuecat_internal_helper_spec.rb

```diff
@@ -593,4 +580,173 @@ def setup_stubs
+      setup_stubs
+      allow(Fastlane::Actions::GithubApiAction).to receive(:run)
+        .with(server_url: server_url,
+              path: '/repos/RevenueCat/mock-repo-name/compare/1.11.0...HEAD',
+              http_method: http_method,
+              body: {},
+              api_token: 'mock-github-token')
+        .and_return(get_commits_response_patch)
+      expect_any_instance_of(Object).not_to receive(:sleep)
+      next_version = Fastlane::Helper::RevenuecatInternalHelper.determine_next_version_using_labels(
+        'mock-repo-name',
+        'mock-github-token',
+        0
+      )
+      expect(next_version).to eq("1.11.1")
+    end
+
+    it 'determines next version as minor correctly' do
+      setup_stubs
+      expect_any_instance_of(Object).not_to receive(:sleep)
+      next_version = Fastlane::Helper::RevenuecatInternalHelper.determine_next_version_using_labels(
+        'mock-repo-name',
+        'mock-github-token',
+        0
+      )
+      expect(next_version).to eq("1.12.0")
+    end
+
+    it 'determines next version as major correctly' do
+      setup_stubs
+      allow(Fastlane::Actions::GithubApiAction).to receive(:run)
+        .with(server_url: server_url,
+              path: '/search/issues?q=repo:RevenueCat/mock-repo-name+is:pr+base:main+SHA:a72c0435ecf71248f311900475e881cc07ac2eaf',
+              http_method: http_method,
+              body: {},
+              api_token: 'mock-github-token')
+        .and_return(get_breaking_commit_response)
+      expect_any_instance_of(Object).not_to receive(:sleep)
+      next_version = Fastlane::Helper::RevenuecatInternalHelper.determine_next_version_using_labels(
+        'mock-repo-name',
+        'mock-github-token',
+        0
+      )
+      expect(next_version).to eq("2.0.0")
+    end
+
+    it 'sleeps between getting commits info if passing rate limit sleep' do
+      setup_stubs
+      expect_any_instance_of(Object).to receive(:sleep).with(3).exactly(3).times
+      next_version = Fastlane::Helper::RevenuecatInternalHelper.determine_next_version_using_labels(
+        'mock-repo-name',
+        'mock-github-token',
+        3
+      )
+      expect(next_version).to eq("1.12.0")
+    end
+
+    it 'fails if it finds multiple commits with same sha' do
+      setup_stubs
+      allow(Fastlane::Actions::GithubApiAction).to receive(:run)
+        .with(server_url: server_url,
+              path: '/search/issues?q=repo:RevenueCat/mock-repo-name+is:pr+base:main+SHA:0e67cdb1c7582ce3e2fd00367acc24db6242c6d6',
+              http_method: http_method,
+              body: {},
+              api_token: 'mock-github-token')
+        .and_return(get_duplicate_items_fix_commit_response)
+      expect do
+        Fastlane::Helper::RevenuecatInternalHelper.determine_next_version_using_labels(
+          'mock-repo-name',
+          'mock-github-token',
+          0
+        )
+      end.to raise_exception(StandardError)
+    end
+
+    def setup_stubs
```

Comment:
Hmm the link doesn't seem to direct to the correct place for me, could you post it again? I think I know what code you're mentioning and I agree it would be ideal to extract it... Maybe we can pass a map of sha's to responses as a parameter to generate this programmatically?

Also, maybe we should rename this to `setup_commit_search_stubs` or something like that? So it's not so generic (I should have noticed that when I did the other tests 😅 )

Reply:
I am talking about the `setup_stubs` function in `auto_generate_changelog`